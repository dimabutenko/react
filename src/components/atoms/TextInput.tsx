import React, { type HTMLInputTypeAttribute, type KeyboardEventHandler } from 'react'

export interface TextInputProps {
  type: HTMLInputTypeAttribute
  onKeyDown?: KeyboardEventHandler
  setInputRef?: React.LegacyRef<HTMLInputElement> | undefined
  placeholder?: string | undefined
  class?: string | undefined
  attributes?: any | undefined
}

const TextInput: React.FC<TextInputProps> = (props: TextInputProps) => (
  <input
    type={props.type}
    ref={props.setInputRef}
    className={props.class}
    placeholder={props.placeholder}
    onKeyDown={props.onKeyDown}
    {...props.attributes}
  />
)

export default TextInput
