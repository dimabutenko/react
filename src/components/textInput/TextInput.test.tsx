import React from 'react'
import { fireEvent, getByTestId, render } from '@testing-library/react'
import TextInput from './TextInput'

describe('TextInput component tests', () => {
  test('Should render correctly', () => {
    const { container } = render(<TextInput type={'text'} />)
    expect(container).toMatchSnapshot()
  })

  test('Should call onKeyDown when KeyDown', () => {
    const onKeyDown = jest.fn()
    const { container } = render(
      <TextInput attributes={{ 'data-testid': 'my-input' }} onKeyDown={onKeyDown} type={'text'} />
    )
    fireEvent.keyDown(getByTestId(container, 'my-input'), { key: 'Enter' })
    expect(onKeyDown).toBeCalled()
  })
})
