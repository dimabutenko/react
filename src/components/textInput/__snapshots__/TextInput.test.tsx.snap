// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component tests Should render correctly 1`] = `
<div>
  <input
    type="text"
  />
</div>
`;
